docs/references/api/resource_embedding.rst

```diff
@@ -318,10 +320,360 @@ Computed relationships have good performance as their intended design enable `in
 
   - Make sure to correctly label the ``to-one`` part of the relationship. When using the ``ROWS 1`` estimation, PostgREST will expect a single row to be returned. If that is not the case, it will unnest the embedding and return repeated values for the top level resource.
 
+.. _embed_disamb:
+.. _hint_disamb:
+.. _target_disamb:
+.. _complex_rels:
+
+Complex Relationships
+=====================
+
+As mentioned on :ref:`resource_embedding`, the server does joins based on **Foreign Keys**.
+When there are many foreign keys between tables, it needs disambiguation to resolve which foreign key columns to use for the join.
+
+:ref:`computed_relationships` can do the job here, they can choose join columns arbitrarily.
+
+.. note::
+
+  `Embedding disambiguation <https://postgrest.org/en/v11.1/references/api/resource_embedding.html#embedding-disambiguation>`_ using ``!hint`` or ``!target`` is now deprecated.
+  Follow the solutions in this section when a ``300 Multiple Choices`` error is returned.
+
+.. _multiple_m2o:
+
+Multiple Many-To-One
+--------------------
+
+.. tabs::
+
+  .. group-tab:: ERD
+
+    .. image:: ../../_static/orders.png
+
+  .. code-tab:: postgresql SQL
+
+    create table addresses (
+      id int primary key generated always as identity,
+      name text,
+      city text,
+      state text,
+      postal_code char(5)
+    );
+
+    create table orders (
+      id int primary key generated always as identity,
+      name text,
+      billing_address_id int references addresses(id),
+      shipping_address_id int references addresses(id)
+    );
+
+To successfully embed ``orders`` with ``addresses``, you need to create computed relationships for the foreign keys columns you want to use:
+
+.. code-block:: postgresql
+
+  create function billing_address(orders) returns setof addresses rows 1 as $$
+    select * from addresses where id = $1.billing_address_id
+  $$ stable language sql;
+
+  create function shipping_address(orders) returns setof addresses rows 1 as $$
+    select * from addresses where id = $1.shipping_address_id
+  $$ stable language sql;
+
+Now, we can unambiguously embed the billing and shipping addresses by selecting the computed relationships.
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /orders?select=name,billing_address(name),shipping_address(name) HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/orders?select=name,billing_address(name),shipping_address(name)"
+
+.. code-block:: json
+
+   [
+     {
+       "name": "Personal Water Filter",
+       "billing_address": {
+         "name": "32 Glenlake Dr.Dearborn, MI 48124"
+       },
+       "shipping_address": {
+         "name": "30 Glenlake Dr.Dearborn, MI 48124"
+       }
+     }
+   ]
+
+.. _multiple_o2m:
+
+Multiple One-To-Many
+--------------------
```

Comment:
Since we're looking for completeness in the examples, I added this one too. We could remove it if it's too much.

Reply:
Looks good!